lib/puppet_x/firewalld/property/rich_rule_action.rb

```diff
@@ -7,7 +7,7 @@ def _validate_action(value)
         end
         validate do |value|
           if value.is_a?(Hash)
-            if value.keys.sort != %i[action type]
+            if value.keys.sort != [:action, :type]
```

Comment:
This is correct.  I've cherry-picked your first commit into the new PR.  Thanks!